src/main/java/com/juniorstart/juniorstart/service/ProjectService.java

```diff
@@ -0,0 +1,258 @@
+
+        if(projectRepository.findByName(projectRequest.getName()).isPresent()) {
+            throw new BadRequestException("Name already in use.");
+        }
+
+        Project newProject = createProjectModel(projectRequest);
+
+        return projectRepository.save(newProject);
+
+    }
+
+    /**
+     * Update a existing project.
+     * @param currentUser A current user's credentials.
+     * @param projectId A project identification number.
+     * @param projectRequest A request thar contains a credentials for a update project.
+     * @return a updated project.
+     * @throws BadRequestException if user hasn't has a ownership of project.
+     */
+    public ProjectResponse updateProject(UserPrincipal currentUser, Long projectId, ProjectRequest projectRequest) {
+        User user = userDao.findById(currentUser.getId()).orElseThrow(() -> new ResourceNotFoundException("User", "userId", currentUser.getId()));
+
+        Project currentProject = projectRepository.findById(projectId).orElseThrow(() -> new ResourceNotFoundException("Project", "id", projectId));
+
+        long currentUserIdentification = user.getPublicId();
+        long projectOwnerIdentification = currentProject.getCreatedBy();
+
+        boolean isCurrentUserIsOwnerOfProject = checkIsUserCreatedProject(currentProject, user);
+
+        if(isCurrentUserIsOwnerOfProject) {
+            currentProject.setRecruiting(projectRequest.isRecruiting());
+            currentProject.setBody(projectRequest.getBody());
+            currentProject.setDescription(projectRequest.getDescription());
+            currentProject.setNumberOfSeats(projectRequest.getNumberOfSeats());
+            currentProject.setTitle(projectRequest.getTitle());
+            currentProject.setName(projectRequest.getName());
+            currentProject.setRepository(projectRequest.getRepository());
+
+            Project updatedProject = projectRepository.save(currentProject);
+
+
+
+            return ModelMapper.mapProjectToProjectResponse(currentProject, user);
+        } else {
+            throw new BadRequestException("You are not a owner of this project!");
+        }
+
+    }
+
+    /**
+     * Delete a project
+     * @param currentUser A current user's credentials.
+     * @param projectId A project's identification number.
+     * @return a {@link ResponseEntity} with the message.
+     * @throws BadRequestException if user hasn't created a this project.
+     */
+    public ResponseEntity<?> deleteProject(UserPrincipal currentUser, Long projectId) {
+        Project currentProject = projectRepository.findById(projectId).orElseThrow(() -> new ResourceNotFoundException("Project", "projectId", projectId));
+
+        User currentLoggedUser = userDao.findByPrivateId(currentUser.getId()).orElseThrow(() -> new ResourceNotFoundException("User", "userId", currentProject.getId()));
```

Comment:
Refaktoryzacja

Reply:
pominięte

Reply:
Nie wiem w jaki sposób mam jeszcze bardziej zrefaktorować.

Reply:
Zgodnie  z DRY nie powtarzasz kodu tylko go wyciągasz do metody pomocniczej. 
projectRepository.findById występuje u Cb 3 razy. 
projectRepository.findByI 3 razy. 

    private Project getCurrentProject(Long projectId) {
        return projectRepository.findById(projectId).orElseThrow(() -> new ResourceNotFoundException("Project", "id", projectId));
    }

I odnosisz się tylko do metody. Powiesz po co, a no po to że jak coś będziesz musiał tutaj zmienić to zmieniasz w 3 miejscach zamiast w 1 miejscu. 
Dodatkowo jak masz wysoce złożoną metodę to ją dzielisz na mniejsze części. Sprawia to że kod jest o wiele czytelniejszy.